src/edu/stanford/nlp/sempre/DateValue.java

```diff
@@ -51,24 +66,56 @@ private static int parseIntRobust(String i) {
     return val;
   }
 
+  private static double parseDoubleRobust(String i) {
+    try {
+      return Double.parseDouble(i);
+    } catch (NumberFormatException ex) {
+      return 0;
+    }
+  }
+
   public static DateValue now() {
     Calendar cal = Calendar.getInstance();
     int year = cal.get(Calendar.YEAR);
     int month = cal.get(Calendar.MONTH);
     int day = cal.get(Calendar.DAY_OF_MONTH);
-    return new DateValue(year, month, day);
+    int hour = cal.get(Calendar.HOUR_OF_DAY);
+    int minute = cal.get(Calendar.MINUTE);
+    double second = cal.get(Calendar.SECOND) + 0.001 * cal.get(Calendar.MILLISECOND);
+    return new DateValue(year, month, day, hour, minute, second);
   }
 
   public DateValue(int year, int month, int day) {
     this.year = year;
     this.month = month;
     this.day = day;
+    this.hour = 0;
+    this.minute = 0;
+    this.second = 0;
+  }
+
+  public DateValue(int year, int month, int day, int hour, int minute, double second) {
+    this.year = year;
+    this.month = month;
+    this.day = day;
+    this.hour = hour;
+    this.minute = minute;
+    this.second = second;
   }
 
   public DateValue(LispTree tree) {
     this.year = Integer.valueOf(tree.child(1).value);
     this.month = Integer.valueOf(tree.child(2).value);
     this.day = Integer.valueOf(tree.child(3).value);
+    if (tree.children.size() > 3) {
+      this.hour = Integer.valueOf(tree.child(4).value);
+      this.minute = Integer.valueOf(tree.child(5).value);
+      this.second = Double.valueOf(tree.child(6).value);
+    } else {
+      this.hour = 0;
```

Comment:
-1 (actually, probably would be a good idea to define a constant INVALID = -1 and just use that everywhere)